### PR TITLE
Add new AUTHENTICATION action type.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Task Issue:
- https://github.com/wso2/product-is/issues/20931

Following improvements are done with this PR:
- Add new action type "AUTHENTICATION"

Related PR:
- https://github.com/wso2/identity-api-server/pull/646

### CODE REVIEW

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java [hunk 1]
```diff
@@ -65,6 +65,8 @@ public boolean isExecutionForActionTypeEnabled(ActionType actionType) {
         switch (actionType) {
             case PRE_ISSUE_ACCESS_TOKEN:
                 return isActionTypeEnabled(ActionTypeConfig.PRE_ISSUE_ACCESS_TOKEN.getActionTypeEnableProperty());
+            case AUTHENTICATION:
+                return isActionTypeEnabled(ActionTypeConfig.AUTHENTICATION.getActionTypeEnableProperty());
```
Comment (ashanthamara) [inline components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java:69]: Let's add the config to the identity.xml.j2 and the default.json as well.

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java [hunk 2]
```diff
@@ -92,13 +94,18 @@ private boolean isActionTypeEnabled(String actionTypePropertyName) {
...
             case PRE_ISSUE_ACCESS_TOKEN:
-                List<String> excludedHeadersPropertyValue = getPropertyValues(
+                excludedHeadersPropertyValue = getPropertyValues(
                         ActionTypeConfig.PRE_ISSUE_ACCESS_TOKEN.getExcludedHeadersProperty());
                 excludedHeaders.addAll(excludedHeadersPropertyValue);
                 break;
+            case AUTHENTICATION:
+                excludedHeadersPropertyValue = getPropertyValues(
+                        ActionTypeConfig.AUTHENTICATION.getExcludedHeadersProperty());
+                excludedHeaders.addAll(excludedHeadersPropertyValue);
```
Comment (ashanthamara) [inline components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java:107]: Shall we move this out of the switch case

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java [hunk 3]
```diff
@@ -124,13 +131,18 @@ private Set<String> getExcludedHeadersInActionRequestForAllTypes() {
...
             case PRE_ISSUE_ACCESS_TOKEN:
-                List<String> excludedParamsPropertyValue = getPropertyValues(
+                excludedParamsPropertyValue = getPropertyValues(
                         ActionTypeConfig.PRE_ISSUE_ACCESS_TOKEN.getExcludedParamsProperty());
                 excludedParams.addAll(excludedParamsPropertyValue);
                 break;
+            case AUTHENTICATION:
+                excludedParamsPropertyValue = getPropertyValues(
+                        ActionTypeConfig.AUTHENTICATION.getExcludedParamsProperty());
+                excludedParams.addAll(excludedParamsPropertyValue);
```
Comment (ashanthamara) [inline components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java:144]: Same as above comment

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java:66]: Addressed with commit 2791587f98eb05e7560e1de7950fb41ca23d1218

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java (hunk 3 above)
Comment (Thisara-Welmilla) [reply components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java]: Addressed with commit 2791587f98eb05e7560e1de7950fb41ca23d1218

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java]: Addressed with commit 2791587f98eb05e7560e1de7950fb41ca23d1218

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 4]
```diff
@@ -1605,7 +1605,11 @@
     "client_assertion_type",
     "client_assertion"
   ],
-
+  "actions.types.authentication.enable": true,
+  "actions.types.authentication.action_request.excluded_headers": [
```
Comment (Thisara-Welmilla) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:1609]: Remove empty list.

Comment (Thisara-Welmilla) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: Addressed with commit e4ed22cd5d30c42f262592d3c61ce513e8bd0e17

**Pull request merged**